utils/download_datasets_and_models.py

```diff
@@ -0,0 +1,256 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import array
+import collections
+import gzip
+import os.path
+import pickle
+import sys
+import tarfile
+import urllib
+import sys
+
+try:
+    from urllib.error import URLError
+except ImportError:
+    from urllib2 import URLError
+
+try:
+    from urllib.request import urlretrieve
+except ImportError:
+    from urllib import urlretrieve
+
+Dataset = collections.namedtuple('TargetItem', 'filename, url, handler, dest_path')
+
+
+def pickle_load(file):
```

Comment:
Comment.

Reply:
Load a file using pickle module, and parameters vary based on different Python versions.

Comment added.